### PR TITLE
Api location filters

### PR DESCRIPTION
**Pull request by fm1psy**
Refactored code to have a location function that then calls the country or continent filter based on the inputs given.
This makes the function getting all the earthquake data easier to read and understand.

### CODE REVIEW

**Review by joe1606 — CHANGES_REQUESTED** [review]
i think a conftest file would be good

File: api/api.py [hunk 1]
```diff
@@ -60,17 +62,18 @@ def get_filter_queries(earthquake_filters: dict[str]) -> list[str]:
...
+    if magtype not in FILTER_MISINPUTS:
         res.append(f"mt.magtype_value = '{magtype}'")
-    if event is not None:
+    if event not in FILTER_MISINPUTS:
         res.append(f"t.type_value = '{event}'")
-    if min_magnitude is not None:
+    if min_magnitude not in FILTER_MISINPUTS:
         res.append(f"e.magnitude >= '{min_magnitude}'")
     if res:
         res[0] = "WHERE "+res[0]
```
Comment (laUmar123) [inline api/api.py:79]: I believe this snippet of code would look better like this:
```
res = []
    if earthquake_filters[STATUS_FILTER_KEY] not in FILTER_MISINPUTS:
        res.append(f"s.status = '{earthquake_filters[STATUS_FILTER_KEY]}'")
    if earthquake_filters[NETWORK_FILTER_KEY] not in FILTER_MISINPUTS:
        res.append(f"n.network_name = '{earthquake_filters[NETWORK_FILTER_KEY]}'")
    if earthquake_filters[ALERT_FILTER_KEY] not in FILTER_MISINPUTS:
        res.append(f"a.alert_value = '{earthquake_filters[ALERT_FILTER_KEY]}'")
    if earthquake_filters[MAG_TYPE_FILTER_KEY] not in FILTER_MISINPUTS:
        res.append(f"mt.magtype_value = '{earthquake_filters[MAG_TYPE_FILTER_KEY]}'")
    if earthquake_filters[EVENT_FILTER_KEY] not in FILTER_MISINPUTS:
        res.append(f"t.type_value = '{earthquake_filters[EVENT_FILTER_KEY]}'")
    if earthquake_filters[MIN_MAGNITUDE_FILTER_KEY] not in FILTER_MISINPUTS:
        res.append(f"e.magnitude >= {earthquake_filters[MIN_MAGNITUDE_FILTER_KEY]}")
    if res:
        res[0] = "WHERE " + res[0]
    return res
```
what do you think?

File: api/api.py [hunk 2]
```diff
@@ -173,7 +183,7 @@ def get_earthquakes() -> Response:
             COUNTRY_FILTER_KEY: request.args.get("country")
         }
 
-        earthquakes = get_all_earthquakes(user_filters)
+        earthquakes = get_earthquake_data(user_filters)
         return earthquakes, 200
```
Comment (laUmar123) [inline api/api.py:187]: Please change this `return jsonify(earthquakes), 200`

**Review by laUmar123 — CHANGES_REQUESTED** [review]
Nice work, just 2 things I'd suggest changing

**Review by laUmar123 — APPROVED** [review]

File: api/api.py (hunk 1 above)
Comment (fm1psy) [reply api/api.py:79]: The reason I use variables here is to make it more legible. It gives each value an easier name to read, and makes the code less cluttered in my opinion.

**Review by joe1606 — APPROVED** [review]

**Pull request merged**